codec/amino_codec.go

```diff
@@ -78,3 +82,24 @@ func (ac *AminoCodec) UnmarshalJSON(bz []byte, ptr proto.Message) error {
 func (ac *AminoCodec) MustUnmarshalJSON(bz []byte, ptr proto.Message) {
 	ac.LegacyAmino.MustUnmarshalJSON(bz, ptr)
 }
+
+// MarshalInterface implements BinaryMarshaler interface
+// The `o` must be an interface and must must implement ProtoMarshaler (it will panic otherwise).
```

Comment:
why do we need a ProtoMarshaler actually?

Reply:
It should be `proto.Message`, error in the comment. Thanks.